Framework/TestHelpers/src/MuonWorkspaceCreationHelper.cpp

```diff
@@ -203,6 +203,32 @@ ITableWorkspace_sptr createDeadTimeTable(const size_t &nspec,
   return deadTimeTable;
 }
 
+/**
+ * Create a simple time zero TableWorkspace with one column (time zero)
+ * @param numSpec :: The number of spectra (rows of the table)
+ * @param timeZeros :: Vector of time zeros for each spectra
+ * @return TableWorkspace with time zeros in each row for all spectra
+ */
+ITableWorkspace_sptr createTimeZeroTable(const size_t &numSpec,
+                                         std::vector<double> &timeZeros) {
+
+  auto timeZeroTable = std::dynamic_pointer_cast<ITableWorkspace>(
+      WorkspaceFactory::Instance().createTable("TableWorkspace"));
+
+  timeZeroTable->addColumn("double", "time zero");
+
+  if (timeZeros.size() != numSpec) {
+    return timeZeroTable;
```

Comment:
You need to log a warning rather than silently returning nothing, lest our support people start to hate you when they get reports that empty tables are being produced for some reason ;)

Reply:
Does this need an instance of the logger here? as it should just be used for unit tests and not actually seen by users?

Reply:
It's genuinely your call: I didn't see the path (in TestHelpers), but the metric I personally use is: "If I added an error message in the logger, would I thank myself when trying to diagnose a failed test".

Sometimes the answer is no as the test makes it painfully obvious without the logger, sometimes it would save me having to break out a debugger blind